software/webapp/src/main/java/brooklyn/entity/webapp/jboss/Wildfly8SshDriver.java

```diff
@@ -0,0 +1,249 @@
+            getMachine().copyTo(keystoreStream, destinationSslKeystoreFile);
+        }
+
+        // Copy the configuration file across
+        String destinationConfigFile = Os.mergePathsUnix(getRunDir(), SERVER_TYPE, "configuration", CONFIG_FILE);
+        copyTemplate(getTemplateConfigurationUrl(), destinationConfigFile);
+
+        // Copy the initial wars to the deploys directory
+        getEntity().deployInitialWars();
+    }
+
+    @Override
+    public void launch() {
+        entity.setAttribute(Wildfly8Server.PID_FILE, Os.mergePathsUnix(getRunDir(), PID_FILENAME));
+
+        // We wait for evidence of JBoss running because, using
+        // brooklyn.ssh.config.tool.class=brooklyn.util.internal.ssh.cli.SshCliTool,
+        // we saw the ssh session return before the JBoss process was fully running
+        // so the process failed to start.
+        newScript(MutableMap.of(USE_PID_FILE, false), LAUNCHING)
+                .body.append(
+                        "export LAUNCH_JBOSS_IN_BACKGROUND=true",
+                        format("export JBOSS_HOME=%s", getExpandedInstallDir()),
+                        format("export JBOSS_PIDFILE=%s/%s", getRunDir(), PID_FILENAME),
+                        format("%s/bin/%s.sh ", getExpandedInstallDir(), SERVER_TYPE) +
+                                format("--server-config %s ", CONFIG_FILE) +
+                                format("-Djboss.server.base.dir=%s/%s ", getRunDir(), SERVER_TYPE) +
+                                format("\"-Djboss.server.base.url=file://%s/%s\" ", getRunDir(), SERVER_TYPE) +
+                                "-Djava.net.preferIPv4Stack=true " +
+                                "-Djava.net.preferIPv6Addresses=false " +
+                                format(" >> %s/console 2>&1 </dev/null &", getRunDir()),
+                        "for i in {1..10}\n" +
+                                "do\n" +
+                                "    grep -i 'starting' "+getRunDir()+"/console && exit\n" +
+                                "    sleep 1\n" +
+                                "done\n" +
+                                "echo \"Couldn't determine if process is running (console output does not contain 'starting'); continuing but may subsequently fail\""
+                    )
+                .execute();
+    }
+
+    @Override
+    public boolean isRunning() {
+        return newScript(MutableMap.of(USE_PID_FILE, true), CHECK_RUNNING).execute() == 0;
+    }
+
+    @Override
+    public void stop() {
+        newScript(MutableMap.of(USE_PID_FILE, true), STOPPING).environmentVariablesReset().execute();
+    }
+
+    @Override
+    public void kill() {
+        newScript(MutableMap.of(USE_PID_FILE, true), KILLING).execute();
+    }
+
+    @Override
+    protected List<String> getCustomJavaConfigOptions() {
+        return MutableList.<String>builder()
+                .addAll(super.getCustomJavaConfigOptions())
+                .add("-Xms200m")
```

Comment:
I'm fine with this for now because it's the same as JBoss7Server...

However, would be good if this was easier to control via config. If someone wanted a different Xmx then they'd currently (probably?) have to write code to override this, rather than being able to do it in YAML.

For example, maybe we should look at using `UsesJava.JAVA_OPTS` (here, in Tomcat, and in a whole bunch of other entities).

Let's not do that in this PR though!